### PR TITLE
Fix rustdoc help options

### LINKED ISSUE #98976: rustdoc -Z help and -C help are still broken
I tried this: `rustdoc -Zhelp`

I expected to see this happen: A list of unstable options

Instead, this happened:
```
$ rustdoc +stage1 -Zhelp
error: unknown unstable option: `help`

$ rustdoc +stage1 -Chelp
error: unknown codegen option: `help`
```

The options need to be built *after* checking for help, not before.
https://github.com/rust-lang/rust/blob/0567674460504d916c254696e1b25082f72a66bf/src/librustdoc/config.rs#L338-L355

cc @GuillaumeGomez

### Meta

`rustc --version --verbose`: 0567674460504d916c254696e1b25082f72a66bf

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #98976.

Since you're the one who found out about the problem and also provided the solution (thanks for both!):

r? @jyn514

### CODE REVIEW

**Review by jyn514 — APPROVED** [review]

File: src/test/rustdoc-ui/c-help.rs [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// check-pass
+// compile-flags: -Chelp
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/test/rustdoc-ui/c-help.rs:2]: Excellent idea!

**Comment by GuillaumeGomez** [comment]
Updated.

**Comment by GuillaumeGomez** [comment]
Strange that it doesn't find the pattern. I'll check what I did wrong.

**Comment by GuillaumeGomez** [comment]
I needed to add a new check (`regex-error-pattern`) to allow regex checks. Should be fixed now.

**Comment by jyn514** [comment]
I don't think we should add regex support for compiletest just for this. You can use a different pattern for error-pattern instead, `incremental=val` or something like that.

**Comment by GuillaumeGomez** [comment]
I thought the idea was to confirm that we had `-Z`/`-C` mostly. For example, it'd be possible to have `-C incremental=val` whereas `-Z` was used (proving that the wrong call is being performed in rustdoc). Don't know if that makes sense or if I'm being too cautious here?

**Comment by jyn514** [comment]
Hmm, fair enough. cc @Mark-Simulacrum just in case you happen to have strong opinions about adding a regex matcher to the UI test runner, but this seems fine to me.

@bors r+

**Comment by matthiaskrgr** [comment]
@bors r- 
failed in a rollup: https://github.com/rust-lang/rust/pull/99127#issuecomment-1179807584

**Comment by GuillaumeGomez** [comment]
Updated the UI output.

@bors r=jyn514

**Pull request merged**